### PR TITLE
Added JSON-RPC method class and definitions

### PR DESCRIPTION
**Pull request by sterlingwellscaffeine**
## Purpose

1.) Added generic method class and defined all JSON-RPC routes

## Approach

1.) A generic module (in this case the requests module) will handle the calls done by the method class. 
I want to expand this class with formatters and validators but will do that at a later point, for now it is very minimal. 

## Testing

Unit testing of the method class.

### CODE REVIEW

File: brambl/Brambl.py [hunk 1]
```diff
@@ -1,172 +1,22 @@
...
-
-from brambl.utils.conversions import to_bytes, Primitives, to_text, to_hex, to_base58
+from brambl.client.rpc import HTTPClient
+from brambl.typing.encoding import HexStr, Base58Str
+from brambl.utils.conversions import to_bytes, Primitives, to_text, to_hex, to_int, to_base58
 from brambl.utils.encoding import to_json
+from brambl.manager import RequestManager as DefaultRequestManager
 
 # Constants definitions
 validTxMethods = ['createAssetsPrototype', 'transferAssetsPrototype', 'transferTargetAssetsPrototype']
```
Comment (scasplte2) [inline brambl/Brambl.py:11]: do you use this anywhere? the targetAsset thing is deprecated so I'm pretty sure this line can be removed

File: brambl/Brambl.py [hunk 2]
```diff
@@ -1,172 +1,22 @@
...
 
 
 class Brambl:
```
Comment (scasplte2) [inline brambl/Brambl.py:14]: perhaps I don't read python well but it seems like this is a very different scheme for Brambl organization from at least JS. In that one the Brambl instance is the key object that everything flows through. Is that still the case here?

**Review by scasplte2 — CHANGES_REQUESTED** [review]
overall nicely done, I like the structure of the RPC. If you can address my two comments we can merge

File: brambl/Brambl.py (hunk 1 above)
Comment (sterlingwellscaffeine) [reply brambl/Brambl.py:11]: No, that is a hold over from the code that was previously there. Good catch that will be fixed.

File: brambl/Brambl.py (hunk 2 above)
Comment (sterlingwellscaffeine) [reply brambl/Brambl.py:14]: It is more similar to the "Requests" class in JS. Potentially I should add the CredentialManager to the Brambl instance as well?

File: brambl/Brambl.py (hunk 1 above)
Comment (sterlingwellscaffeine) [reply brambl/Brambl.py]: Fixed in most recent commit.

File: brambl/Brambl.py (hunk 2 above)
Comment (scasplte2) [reply brambl/Brambl.py:23]: maybe we can just talk about it today but I'd like to discuss the entire structure so it is clear in my head

Comment (sterlingwellscaffeine) [reply brambl/Brambl.py:23]: Yes, we should definitely do some knowledge transfer as well when we get an opportunity.

**Pull request merged**